luigi/contrib/dropbox.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019 Jose-Ignacio Riaño Chico
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+#
+
+from __future__ import absolute_import
+
+import datetime
+import logging
+import ntpath
+import os
+import random
+import tempfile
+from contextlib import contextmanager
+
+import luigi.format
+from luigi.target import FileSystem, FileSystemTarget, AtomicLocalFile
+
+logger = logging.getLogger('luigi-interface')
+
+try:
+    import dropbox.dropbox
+    import dropbox.exceptions
+    import dropbox.files
+except ImportError:
+    logger.warning("Loading Dropbox module without the python package dropbox (https://pypi.org/project/dropbox/). "
+                   "Will crash at runtime if Dropbox functionality is used.")
+
+
+class DropboxClient(FileSystem):
+    """
+    Dropbox client for authentication, designed to be used by the :py:class:`DropboxTarget` class.
+    """
+
+    def __init__(self, token):
+        """
+        :param str token: Dropbox Oauth2 Token. See :class:`DropboxTarget` for more information about generating a token
+        """
+        if not token:
+            raise ValueError("The token parameter must contain a valid Dropbox Oauth2 Token")
+
+        try:
+            conn = dropbox.dropbox.Dropbox(oauth2_access_token=token, user_agent="Luigi")
+        except Exception as e:
+            raise Exception("Cannot connect to Dropbox. Check your Internet connection and the token. \n" + repr(e))
+
+        self.token = token
+        self.conn = conn
+
+    def exists(self, path):
+        if path == '/':
+            return True
+        try:
+            md = self.conn.files_get_metadata(path)
+            return bool(md)
```

Comment:
wouldn't this line only ever return true?

Reply:
You are right. I have changed it.